README.md

```diff
@@ -673,6 +674,36 @@ type FnReturnType = $Call<ExtractReturnType<Fn>>; // number
 
 [⇧ back to top](#flows-utility-types)
 
+### `$ObjMap<T, F>`
+
+takes an object type `T`, and a function type `F`, and returns the object type obtained by mapping the type of each value in the object with the provided function type `F`
+https://flow.org/en/docs/types/utilities/#toc-objmap
+
+**Usage:**
+
+```ts
+import { $ObjMap } from 'utility-types';
+
+type Obj = {
+  a: number;
+  b?: boolean;
+  c: string | null;
+};
+type Mapper = (value: number | boolean | string | null) => string;
```

Comment:
That looks interesting, but I would really like to solve below case with this type:
```
type Obj = {
      a: () => number;
      b: () => boolean;
      c: () => string | null;
    };

    type Mapper = <T>(value: () => T) => T; // the whole point is to use generics in the mapper, but Flow inference works different than in TS

    type MappedObj = $ObjMap<Obj, Mapper>;
//  type MappedObj = {
//    a: number
//    b: boolean
//    c: string|null
//  }
```

Currently it doesn't work and I don't see a way to do that. Any ideas?

Reply:
I tried this line and it works:

```ts
type Mapper = <T>(value: () => T extends any ? T : never) => T;
```

Reply:
I make a bad case demo:

```ts
const f = {} as <T>(value: () => T) => void;
const f1: (value: (() => number) | (() => boolean)) => void = f; // this line failed check
const f2: (value: (() => number | boolean)) => void = f;
```

But I'm not sure why it happened.

Reply:
The problem is the resulting type is not correct. It's falling back to any ({}), please take a look:
![screen shot 2018-12-19 at 10 36 12 am](https://user-images.githubusercontent.com/739075/50211926-18e40180-037a-11e9-8b94-43475e187252.png)

Reply:
There are two probleams.

1. The `Mapper` failed because it doesn't match `F extends _ObjMapper<T>`. I don't knwon why and given a demo.

![image](https://user-images.githubusercontent.com/1812118/50213908-efe36100-03b9-11e9-9422-a2de104d123a.png)

If you use `F extends (...args: any[]) => any` it will work:

![image](https://user-images.githubusercontent.com/1812118/50213938-0689b800-03ba-11e9-9aed-8c6e6d1bfab1.png)

2. The return value should be `{}` ( as known as `any` ) because `ReturnType<Mapper>` returns `{}` in current implement.

There is no `$Call` in TypeScript ( the one in this repo is not fully equal to Flow Type's ). Consider [this Flow Type example](https://flow.org/try/#0C4TwDgpgBAsghmSAnKBeKAeAKgPgBQBucANgK4QBcUeAlGjlFnag1gNwBQokUAShAGdSxYGigASAMIliGeIghIANNWYMAdqQC2AI0U42UAPRG+g4aICWAqJt2KOAYwD26gaKTmRVfkJFiARjYgA):

```js
type Mapper = <T>(value: () => T) => T;
type Result = $Call<Mapper, () => number>; // Result is number
const result: Result = 1;
```

`$Call` can infer `T` because `value` is `() => number` but not TypeScript.

Otherwise you can refactor `$ObjMap` to:

```ts
export type $ObjMap<T extends object, F extends _ObjMapper<T>> = {
  [K in keyof T]: $Call<F, T[K]>; // doesn't work yet
};
```

Reply:
As I said I don't see any way that would work, that's why this PR doesn't make sense because it's basically useless because it doesn't solve the primary use case and is confusing, because people will try to use it and complain it doesn't work in the same way I just did.

I'm afraid I have to reject this.